ethereum/jsonrpc/src/main/java/tech/pegasys/pantheon/ethereum/jsonrpc/websocket/subscription/blockheaders/NewBlockHeadersSubscriptionService.java

```diff
@@ -38,20 +40,23 @@ public NewBlockHeadersSubscriptionService(
 
   @Override
   public void onBlockAdded(final BlockAddedEvent event, final Blockchain blockchain) {
-    final List<NewBlockHeadersSubscription> subscribers =
-        subscriptionManager.subscriptionsOfType(
-            SubscriptionType.NEW_BLOCK_HEADERS, NewBlockHeadersSubscription.class);
+    final ExecutorService executor = Executors.newSingleThreadExecutor();
```

Comment:
We shouldn't create a new executor for every block - need to have this `ExecutorService` managed as part of the component life cycle - started when JSON-RPC starts and shutdown when it's shutdown.  Probably simplest to use vertx - either by jumping to one of it's worker queues or onto its event queue.